File: Solutions/25. Maximum Subarray/sol.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cur_max, total_max = 0, -100001
        for c in nums:
            cur_max = max(c, cur_max + c)
            total_max = max(total_max, cur_max)
        return total_max

File: Solutions/8. Binary Search/sol.cpp
class Solution {
public:
    int search(vector<int>& nums, int target) {
        int l = 0 , r = nums.size() - 1 ;
        while(l < r)
        {
            int m = (l + r)>>1 ;
            if(nums[m] < target)
            {
                l = m + 1 ;
            }
            else
            {
                r = m ;
            }
        }
        if(nums[l] != target)
        {
            return -1 ;
        }
        return l ;
    }
};